src/test/java/com/googleinterns/zoomtube/servlets/LectureListServletTest.java

```diff
@@ -0,0 +1,110 @@
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.reflect.TypeToken;
+import com.googleinterns.zoomtube.data.Lecture;
+import com.googleinterns.zoomtube.utils.LectureUtil;
+import com.ryanharter.auto.value.gson.GenerateTypeAdapter;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+@RunWith(JUnit4.class)
+public final class LectureListServletTest {
+  @Rule public final MockitoRule mockito = MockitoJUnit.rule();
+  @Mock private HttpServletRequest request;
+  @Mock private HttpServletResponse response;
+
+  private final LocalServiceTestHelper testServices =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+  private DatastoreService datastoreService;
+  private LectureListServlet servlet;
+
+  private static final String LINK_INPUT = "link-input";
+  private static final String TEST_LINK = "https://www.youtube.com/watch?v=wXhTHyIgQ_U";
+  private static final String TEST_ID = "wXhTHyIgQ_U";
+
+  @Before
+  public void setUp() throws ServletException {
+    testServices.setUp();
+    datastoreService = DatastoreServiceFactory.getDatastoreService();
+    servlet = new LectureListServlet();
+    servlet.init();
+  }
+
+  @After
+  public void tearDown() {
+    testServices.tearDown();
+  }
+
+  @Test
+  public void doGet_emptyDatabase_shouldReturnNoLecture() throws IOException {
+    StringWriter content = new StringWriter();
+    PrintWriter writer = new PrintWriter(content);
+    when(response.getWriter()).thenReturn(writer);
+
+    servlet.doGet(request, response);
+
+    String json = content.toString();
+    assertThat(json).startsWith("[]");
+  }
+
+  @Test
+  public void doGet_oneLectureInDatabase_shouldReturnOneLecture() throws IOException {
+    when(request.getParameter(LINK_INPUT)).thenReturn(TEST_LINK);
+    datastoreService.put(LectureUtil.createEntity(/* lectureName= */ "", TEST_LINK, TEST_ID));
+    StringWriter content = new StringWriter();
```

Comment:
These could be declared at the class level, given they are used in multiple tests.

Reply:
fixed